protocols/relay/src/behaviour/handler.rs

```diff
@@ -380,6 +387,11 @@ pub struct Handler {
     alive_lend_out_substreams: FuturesUnordered<oneshot::Receiver<()>>,
     /// Futures relaying data for circuit between two peers.
     circuits: Futures<(CircuitId, PeerId, Result<(), std::io::Error>)>,
+
+    inbound_requests_futs:
+        FuturesUnordered<BoxFuture<'static, Result<Event, inbound_hop::FatalUpgradeError>>>,
```

Comment:
I am working on a `futures-bounded` crate in https://github.com/libp2p/rust-libp2p/pull/3914 that I expect to be merged soon. Once that is the case, I'd like you to adopt it here as well to ensure that these aren't unbounded.